lib/client/weblogin.go

```diff
@@ -57,10 +58,13 @@ type SSOLoginConsoleReq struct {
 	// RouteToCluster is an optional cluster name to route the response
 	// credentials to.
 	RouteToCluster string
+	// KubernetesCluster is an optional k8s cluster name to route the response
+	// credentials to.
+	KubernetesCluster string
 }
 
-// Check makes sure that the request is valid
-func (r SSOLoginConsoleReq) Check() error {
+// CheckAndSetDefaults makes sure that the request is valid
+func (r *SSOLoginConsoleReq) CheckAndSetDefaults(pg services.ProxyGetter) error {
```

Comment:
This is doing a lot more than what a typical `CheckAndSetDefaults` does.  Best to call it something else.

Also, `CheckKubeCluster` shouldn't be run until after authorization has completed anyway, otherwise it may allow an unauthenticated user to brute-force the name(s) of the existing kube clusters.

Reply:
Good point about brute-forcing cluster names.
I added it here because any post-auth failure in SSO logins doesn't propagate the error back to the client via redirects. As a result, if a user typos the cluster name, they get a generic error message which is confusing.

Removing this validation for now, we may need to improve the post-auth SSO login failures later on.